tests/testsuite/update.rs

```diff
@@ -2064,3 +2112,508 @@ fn update_breaking_specific_packages() {
+    assert_e2e().eq(&lock_file_after, lock_file);
+
+    p.cargo(
+        "update compatible renamed-from non-semver transitive-compatible transitive-incompatible",
+    )
+    .with_stderr_data(str![[r#"
+[UPDATING] `[..]` index
+[LOCKING] 5 packages to latest compatible versions
+[UPDATING] compatible v1.0.0 -> v1.0.1
+[UPDATING] non-semver v1.0.0 -> v1.0.1 (latest: v2.0.0)
+[UPDATING] renamed-from v1.0.0 -> v1.0.1 (latest: v2.0.0)
+[UPDATING] transitive-compatible v1.0.0 -> v1.0.1
+[UPDATING] transitive-incompatible v1.0.0 -> v1.0.1 (latest: v2.0.0)
+
+"#]])
+    .run();
+}
+
+#[cargo_test]
+fn update_breaking_without_lock_file() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("incompatible", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name  =  "foo"
+            version  =  "0.0.1"
+            edition  =  "2015"
+            authors  =  []
+
+            [dependencies]
+            compatible  =  "1.0"  # Comment
+            incompatible  =  "1.0"  # Comment
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("incompatible", "1.0.1").publish();
+
+    Package::new("incompatible", "2.0.0").publish();
+
+    p.cargo("update -Zunstable-options --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr_data(str![[r#"
+[UPDATING] `[..]` index
+[UPGRADING] incompatible ^1.0 -> ^2.0
+[LOCKING] 3 packages to latest compatible versions
+
+"#]])
+        .run();
+}
+
+#[cargo_test]
+fn update_breaking_spec_version() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("incompatible", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name  =  "foo"
+            version  =  "0.0.1"
+            edition  =  "2015"
+            authors  =  []
+
+            [dependencies]
+            compatible  =  "1.0"  # Comment
+            incompatible  =  "1.0"  # Comment
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("incompatible", "1.0.1").publish();
+
+    Package::new("incompatible", "2.0.0").publish();
+
+    // Invalid spec
+    p.cargo("update -Zunstable-options --breaking incompatible@foo")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[ERROR] expected a version like "1.32"
+
+"#]])
+        .run();
+
+    // Spec version not matching our current dependencies
+    p.cargo("update -Zunstable-options --breaking incompatible@2.0.0")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr_data(str![[r#""#]])
+        .run();
```

Comment:
If I run `cargo update clap@5.0.0` in the cargo repo, it fails.  This should do similar.  This can be added to the task list.

Reply:
Tracked elsewhere

Reply:
By the way, this gets improved in #14140, and becomes:

```
        .with_stderr_data(str![[r#"
[ERROR] package ID specification `incompatible@2.0.0` did not match any packages
Did you mean one of these?

  incompatible@1.0.0

"#]])
```